langserver/completion.go

```diff
@@ -155,41 +154,46 @@ func (s *server) completeFunctionName(completions *[]protocol.CompletionItem, lo
 		return err
 	}
 
+	names := make([]string, 0, len(promql.Functions))
 	for name := range promql.Functions {
-		if strings.HasPrefix(strings.ToLower(name), metricName) {
-			item := protocol.CompletionItem{
-				Label:            name,
-				SortText:         "__1__" + name,
-				Kind:             3, //Function
-				InsertTextFormat: 2, //Snippet
-				TextEdit: &protocol.TextEdit{
-					Range:   editRange,
-					NewText: name + "($1)",
-				},
-				Command: &protocol.Command{
-					// This might create problems with non VS Code clients
-					Command: "editor.action.triggerParameterHints",
-				},
-			}
-			*completions = append(*completions, item)
+		names = append(names, strings.ToLower(name))
```

Comment:
There's a bug here (that existed before). The `ToLower` should actually be applied to `metricName` instead of the function name, since PromQL aggregators like `sum` are case insensitive.

Reply:
Actually the `ToLower` should only be used for filtering aggregators, not for filtering functions.

Reply:
okay sounds good. didn't see some comments before my previous commit . i would address them now